src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
+
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_SIZE = 8;
+    private Object[] stack;
+    private int size = 0;
 
+    public MateStack() {
+        stack = new Object[DEFAULT_SIZE];
+        Arrays.fill(stack, null);
+    }
+
+    public void push(T value) {
+        if (size < stack.length) {
+            stack[size] = value;
+            size++;
+        } else {
+            throw new StackOverflowError();
```

Comment:
The current implementation of the push method throws a StackOverflowError when the stack is full. However, a more typical approach for a stack implementation is to dynamically resize the stack array when it reaches its capacity, rather than throwing an error. Consider implementing a method to resize the stack array to accommodate more elements.